src/main/java/net/glowstone/GlowWorld.java

```diff
@@ -1809,13 +1826,25 @@ private void pulseTickMap() {
     public void requestPulse(GlowBlock block, long tickRate) {
         Location target = block.getLocation();
 
-
         if (tickRate > 0)
             tickMap.put(target, tickRate);
         else if (tickMap.containsKey(target))
             tickMap.remove(target);
     }
 
+    /**
+     * Calling this method will request that the block is ticked after the specified
+     * tix delay
+     *
+     * @param block The block to tick.
+     * @param tickRate The tick delay after which to pulse the block.
+     */
+    public void requestSinglePulse(GlowBlock block, long tickDelay) {
+        Location target = block.getLocation();
+
+        singleTickMap.put(target, worldAge + tickDelay);
```

Comment:
It might cause some bugs that we're not aware of, or if we change something about ticks in the future, having a negative tick delay might cause something weird to happen.